lib/resty/jwt.lua

```diff
@@ -824,12 +826,17 @@ function _M.verify_jwt_obj(self, secret, jwt_obj, ...)
   end
 
   -- if jwe, invoked verify jwe
-  if jwt_obj[str_const.header][str_const.enc]  then
+  if jwt_obj.typ == str_const.JWE or (jwt_obj.typ == nil and jwt_obj.internal ~= nil and jwt_obj[str_const.header][str_const.enc]) then
     return verify_jwe_obj(jwt_obj)
   end
 
   local alg = jwt_obj[str_const.header][str_const.alg]
 
+  if alg == nil then
+    jwt_obj[str_const.reason] = "No algorithm supplied"
+    return jwt_obj
+  end
```

Comment:
we should not check like this, there is already corresponding judgement logic below
```
  else
    jwt_obj[str_const.reason] = "Unsupported algorithm " .. alg
  end
```
We just fix the bug when `alg == nil`
```
  else
    jwt_obj[str_const.reason] = "Unsupported algorithm " .. (alg or "nil")
  end
```